cmake/setup_cached_variables.cmake

```diff
@@ -345,6 +346,12 @@ OPTION(DEAL_II_DOXYGEN_USE_MATHJAX
   )
 MARK_AS_ADVANCED(DEAL_II_DOXYGEN_USE_MATHJAX)
 
+OPTION(DEAL_II_COMPILE_EXAMPLES
+  "If set to ON, all configurable example executables will be built and installed as well. If set to OFF, the examples component only installs the source code of example steps."
+  OFF
```

Comment:
Enabling >100 additional top level targets comes with a significant performance hit (not in the configure stage, but afterwards during makefile generation). I will do some benchmarking on a slow nfs file system to cover a frequent worst case scenario.